sdk/program/src/secp256k1_recover.rs

```diff
@@ -0,0 +1,92 @@
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use core::convert::TryFrom;
+use thiserror::Error;
+
+#[derive(Debug, Clone, PartialEq, Eq, Error)]
+pub enum Secp256k1RecoverError {
+    #[error("The digest provided to a secp256k1_recover is invalid")]
+    InvalidDigest,
+    #[error("The recovery_id provided to a secp256k1_recover is invalid")]
+    InvalidRecoveryId,
+    #[error("The signature provided to a secp256k1_recover is invalid")]
+    InvalidSignature,
+}
+
+#[repr(transparent)]
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    BorshSchema,
+    Clone,
+    Copy,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Hash,
+    AbiExample,
+)]
+pub struct Secp256k1Pubkey(pub [u8; 64]);
```

Comment:
more magic numbers, looks like this 64 is used multiple places, be good to define it